src/main/cli/cli.c

```diff
@@ -1276,74 +1282,109 @@ static void cliSerialPassthrough(char *cmdline)
+            if (bauds[i] == 0) {
+                // Set default baud
+                bauds[i] = 57600;
+                isUseDefaultBaud = true;
+            }
+
+            if (!modes[i]) {
+                modes[i] = MODE_RXTX;
+            }
+
+            *port = openSerialPort(portIDs[i], FUNCTION_NONE, NULL, NULL,
+                                            bauds[i], modes[i],
+                                            SERIAL_NOT_INVERTED);
+            if (!*port) {
+                cliPrintLine("could not be opened.");
+                return;
+            }
+
+            if (isUseDefaultBaud) {
+                cliPrintf("opened, default baud = %d.\r\n", bauds[i]);
+            } else {
+                cliPrintf("opened, baud = %d.\r\n", bauds[i]);
+            }
         } else {
-            cliPrintf("already open, baud = %d.\n\r", passThroughPort->baudRate);
-        }
+            *port = portUsage->serialPort;
+            // If the user supplied a mode, override the port's mode, otherwise
+            // leave the mode unchanged. serialPassthrough() handles one-way ports.
+            // Set the baud rate if specified
+            if (bauds[i]) {
+                cliPrintf("already open, setting baud = %d.\n\r", bauds[i]);
+                serialSetBaudRate(*port, bauds[i]);
+            } else {
+                cliPrintf("already open, baud = %d.\n\r", (*port)->baudRate);
+            }
 
-        if (mode && passThroughPort->mode != mode) {
-            cliPrintf("Mode changed from %d to %d.\r\n",
-                   passThroughPort->mode, mode);
-            serialSetMode(passThroughPort, mode);
-        }
+            if (modes[i] && (*port)->mode != modes[i]) {
+                cliPrintf("Mode changed from %d to %d.\r\n",
+                    (*port)->mode, modes[i]);
+                serialSetMode(*port, modes[i]);
+            }
 
-        // If this port has a rx callback associated we need to remove it now.
-        // Otherwise no data will be pushed in the serial port buffer!
-        if (passThroughPort->rxCallback) {
-            passThroughPort->rxCallback = 0;
+            // If this port has a rx callback associated we need to remove it now.
+            // Otherwise no data will be pushed in the serial port buffer!
+            if ((*port)->rxCallback) {
+                (*port)->rxCallback = 0;
+            }
         }
     }
 
     // If no baud rate is specified allow to be set via USB
-    if (enableBaudCb) {
+    if (leftSidePortID == -1 && enableBaudCb) {
         cliPrintLine("Baud rate change over USB enabled.");
         // Register the right side baud rate setting routine with the left side which allows setting of the UART
         // baud rate over USB without setting it using the serialpassthrough command
-        serialSetBaudRateCb(cliPort, serialSetBaudRate, passThroughPort);
+        serialSetBaudRateCb(*(passThroughPorts[0]), serialSetBaudRate, *(passThroughPorts[1]));
```

Comment:
Have you tested this? What use case do you have in mind for allowing the baud rate to be set through the USB port in the case that the USB port is not involved in the forwarding itself?

I think this functionality should just be declared incompatible with the two-UART forwarding that you are proposing, and treated as such.

Reply:
emm, this code was same as before, I've not test it, just thought it will work fine as before. and In the uartX <-> uartX serial passthrough, it don't need the USB Port to participate the baud rate adjustment, so I add a condition to avoid it(`leftSidePortId == -1`)

Reply:
Ok, I move the 'leftSidePortId == -1` to the rightSideBaud check, look likes:
```
    if (leftSidePortId == -1 && rightSideBaud == 0) {
        enableBaudCb = true;
    }
```